### PR TITLE
[test] Add offline processing integration test

### PR DESCRIPTION
**Pull request by WojciechBarczynski**

### CODE REVIEW

File: integration_tests/src/tests/offline_processing.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    config.queue_options.never_drop_output_frames = true;
+    let instance = CompositorInstance::start(Some(config));
+    let (msg_sender, msg_receiver) = crossbeam_channel::unbounded();
+    start_server_msg_listener(instance.api_port, msg_sender);
+
+    instance.send_request(
+        "input/input_1/register",
+        json!({
+            "type": "mp4",
+            "url":  BUNNY_URL,
```
Comment (WojciechBarczynski) [inline integration_tests/src/tests/offline_processing.rs:30]:
```suggestion
            "url": BUNNY_URL,
```

File: integration_tests/src/tests/offline_processing.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    Command::new("ffprobe")
+        .args(["-v", "error", OUTPUT_FILE])
+        .output()
+        .map_err(|e| anyhow!("Invalid mp4 file. FFprobe error: {}", e))?;
+
+    Ok(())
+}
+
+pub fn start_server_msg_listener(port: u16, event_sender: Sender<tungstenite::Message>) {
```
Comment (wkozyra95) [inline integration_tests/src/tests/offline_processing.rs:108]: can you extract websocket helpers to the crate in a way that can be adapted to use in other tests too?

Comment (wkozyra95) [reply integration_tests/src/tests/offline_processing.rs:108]: I'm not sugesting it but just FYI, as an alternative to the websockets you could subscribe to the events directlly (you will get a channel) 

https://github.com/software-mansion/live-compositor/blob/master/compositor_render/src/event_handler.rs

File: integration_tests/src/tests/offline_processing.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    for msg in msg_receiver.iter() {
+        if let tungstenite::Message::Text(msg) = msg {
+            if msg.contains("\"type\":\"OUTPUT_DONE\",\"output_id\":\"output_1\"") {
+                info!("breaking");
+                break;
+            }
+        }
+    }
+
+    Command::new("ffprobe")
```
Comment (wkozyra95) [inline integration_tests/src/tests/offline_processing.rs:100]: Can we get duration or bitrate with regexp from the output 

Sth like this should be easy to match, but I'm not sure if ffmpeg output is consistent enough
```
  Duration: 00:09:56.47, start: 0.000000, bitrate: 2119 kb/s
```

File: integration_tests/src/tests/utils.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
+use crossbeam_channel::Sender;
```
Comment (wkozyra95) [inline integration_tests/src/tests/utils.rs:1]: move that somwhere else, lets keep tests directory only for tests

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**